### PR TITLE
NO-JIRA: Document machine configuration in HyperShift

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Adds a document explaining how machines are configured in HyperShift

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: docs/content/how-to/configure-machines.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Configuring Machines in HyperShift
+
+In standalone OpenShift, machine configuration is managed via resources like MachineConfig, KubeletConfig, and ContainerRuntimeConfig inside the cluster, and then applied to a set of machines via a MachineConfigPool. In HyperShift clusters, machine configuration is not managed from within a hosted cluster, but rather via the NodePool API outside the cluster.
```
Comment (enxebre) [inline docs/content/how-to/configure-machines.md:3]: may want to include this section into https://hypershift-docs.netlify.app/how-to/automated-machine-management/

Comment (csrwng) [reply docs/content/how-to/configure-machines.md]: Thx! done

File: docs/content/how-to/automated-machine-management/configure-machines.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Configuring Machines in HyperShift
+
+In standalone OpenShift, machine configuration is managed via resources like MachineConfig, KubeletConfig, and ContainerRuntimeConfig inside the cluster, and then applied to a set of machines via a MachineConfigPool. In HyperShift clusters, machine configuration is not managed from within a hosted cluster, but rather via the NodePool API outside the cluster.
```
Comment (sjenning) [inline docs/content/how-to/automated-machine-management/configure-machines.md:3]: s/the NodePool API outside the cluster/a NodePool resource in the hosting cluster/

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Comment by bryan-cox** [comment]
/retitle NO-JIRA: Document machine configuration in HyperShift

File: docs/content/how-to/automated-machine-management/configure-machines.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Configuring Machines in HyperShift
+
+In standalone OpenShift, machine configuration is managed via resources like MachineConfig, KubeletConfig, and ContainerRuntimeConfig inside the cluster, and then applied to a set of machines via a MachineConfigPool. In HyperShift clusters, machine configuration is not managed from within a hosted cluster, but rather via the NodePool API outside the cluster. 
+
+The NodePool field `.spec.config` can be populated with a list of references to configmaps that contain MachineConfig, KubeletConfig, or ContainerRuntimeConfig manifests. This configuration is applied to all machines belonging to that NodePool. In other words, in HyperShift, the NodePool replaces the MachineConfigPool, MachineConfig, KubeletConfig, and ContainerRuntimeConfig resources in standalone OpenShift.
```
Comment (sjenning) [inline docs/content/how-to/automated-machine-management/configure-machines.md:5]: > In other words, in HyperShift, the NodePool replaces the MachineConfigPool, MachineConfig, KubeletConfig, and ContainerRuntimeConfig resources in standalone OpenShift.

Not sure if this adds much and is kinda confusing.  Maybe "In other words, in HyperShift, the NodePool replaces the MachineConfigPool, and MachineConfigs, KubeletConfigs, and ContainerRuntimeConfigs are encapsulated in configmaps and referenced by the NodePool config field."

But the previous sentence kinda already said this so :man_shrugging:

**Review by sjenning — COMMENTED** [review]
A couple of comments but a great addition to the docs!

File: docs/content/how-to/automated-machine-management/configure-machines.md (hunk 3 above)
Comment (csrwng) [reply docs/content/how-to/automated-machine-management/configure-machines.md:5]: ack, yup, it's kind of redundant. Removed.

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Pull request merged**